### PR TITLE
Yd wallet

### PR DESCRIPTION
**Pull request by lyddream**
1、create wallet
2、backup keystore

### CODE REVIEW

File: src/common/Loopring/common/validator.js [hunk 1]
```diff
@@ -13,6 +13,7 @@ let handleErrors = (errors, fields)=>{
 };
 
 let validate = (payload)=>{
+  debugger;
```
Comment (hzbdesigner) [inline src/common/Loopring/common/validator.js]: remove debugger

File: src/components/wallet/components/BackupKeystore.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import {Button, Form} from 'antd';
+import icon from '../../../assets/images/icon-backup-wallet.png'
+import {encrypt} from "Loopring/ethereum/account"
+import {addHexPrefix, clearPrefix, toBuffer} from "Loopring/common/formatter"
+
+function BackupKeystore({account}) {
+
+  const download = () =>{
```
Comment (hzbdesigner) [inline src/components/wallet/components/BackupKeystore.js]: move loopring.js's functions from **components** to **models**, for example: encrypt, addHexPrefix,clearPrefix, toBuffer 。

File: src/components/wallet/components/BackupKeystore.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    const password = account['password'];
+    console.log(password);
+    const wallet = account['privateKey'];
+    const element = document.createElement('a');
```
Comment (hzbdesigner) [inline src/components/wallet/components/BackupKeystore.js]: with React, wo dont need to add or move dom node directly.

File: src/components/wallet/components/BackupKeystore.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    console.log(wallet);
+    element.setAttribute('href', 'data:text/plain;charset=utf-8,' + encodeURIComponent(JSON.stringify(encrypt(toBuffer(addHexPrefix(account['privateKey'])),password))));
```
Comment (hzbdesigner) [inline src/components/wallet/components/BackupKeystore.js]: you can use a 'keystorefile' state to store the file info in the redux state

File: src/components/wallet/components/GenerateWallet.js [hunk 5]
```diff
@@ -1,83 +1,92 @@
 import React from 'react';
-import { connect } from 'dva';
-import { Link } from 'dva/router';
-import { Button,Form,Radio,Input,Tabs,Card,Badge,Icon,Modal,Progress } from 'antd';
+import {connect} from 'dva';
+import {Link} from 'dva/router';
+import {Button, Form, Radio, Input, Tabs, Card, Badge, Icon, Modal, Progress} from 'antd';
 import UnlockByMetaMask from './UnlockByMetaMask'
 import UnlockByKeystore from './UnlockByKeystore'
+import {create} from "Loopring/ethereum/account"
```
Comment (hzbdesigner) [inline src/components/wallet/components/GenerateWallet.js]: move this loopring function to model actions

Comment (lyddream) [reply src/components/wallet/components/GenerateWallet.js]: slove

**Review by hzbdesigner — APPROVED** [review]

**Pull request merged**